dev-tools/check_file_permissions.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+set -eu
+
+files_with_wrong_permissions=$(
+    git ls-files --stage . \
+        ':!*.sh' \
+        ':!php-cs-fixer' \
+    | grep -P "100\d\d(1|3|5|7) " \
+    | sort -fh
+)
+
+if [ "$files_with_wrong_permissions" ]
+then
+    printf '\033[97;41mWrong permissions detected:\033[0m\n'
+    e=$(printf '\033')
+    echo "${files_with_wrong_permissions}"
+    exit 3
+fi
+
+if [ -x "php-cs-fixer" ]
```

Comment:
Oh, like this :) - I was try something `git ls-files` but always ending with silly/stupid command.